README.md

```diff
@@ -1,4 +1,223 @@
+```kotlin
+val sideEffect = object : SideEffect<Int, CalculatorAction> {
+    override val name: String = "network logger"
+
+    override fun CoroutineScope.start(
+        input: ReceiveChannel<CalculatorAction>,
+        stateAccessor: StateAccessor<Int>,
+        output: SendChannel<CalculatorAction>,
+        logger: SideEffectLogger
+    ): Job = launch(context = CoroutineName(name)) {
+        for (inputAction in input) {
+            logger.logSideEffectEvent { LogEvent.SideEffectEvent.InputAction(name, inputAction) }
+            if (inputAction is CalculatorAction.Add &&
+                stateAccessor() >= 0) {
+                launch {
+                    val response = makeNetworkCall()
+                    logger.logSideEffectEvent {
+                        LogEvent.SideEffectEvent.Custom(name, "Received network response: $response")
+                    }
+                    if (response == 200) {
+                        val outputAction = CalculatorAction.Deduct(1)
+                        logger.logSideEffectEvent { LogEvent.SideEffectEvent.DispatchingToReducer(name, outputAction) }
+                        output.send(outputAction)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+You must register you `sideEffect` inside `createStore(sideEffects = listOf(sideEffect))`:
+```kotlin
+val storeWithSideEffect = coroutineScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    sideEffects = listOf(sideEffect),
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+When we subscribe to `storeWithSideEffect` and trigger side effect by dispatching right actions -
+network request from side effect will be performed:
+```kotlin
+storeWithSideEffect.subscribe { state -> updateUI(state) }
+storeWithSideEffect.dispatch(CalculatorAction.Deduct(1))
+// Current state will become -1
+storeWithSideEffect.dispatch(CalculatorAction.Add(11))
+// Current state will become 10 and network request will happen
+```
+
+`CoRedux` also provides two simplified implementations of `SideEffect` that you might find useful:
+- `SimpleSideEffect` that produces on one input action either one output action or no action.
+- `CancellableSideEffect` that cancels previously running `Job` and starts a new one if the same type of action is dispatched to the store.
+
+### Implementation details
+
+With default values in `createStore()`, CoRedux starts main reducer coroutine ("manager"), when first `StateReceiver` subscribes to `CoreduxStore` instance.
```

Comment:
Isn't it repeating  / very similar what you already said before in 

> With default parameters (`launchMode = CoroutineStart.LAZY`), on first subscription, `store` will emit `initialState` to it.

Again, maybe the whole thing of launchMode, StateReceiver and "manager" coroutine should be simplified and put into one place?

Could a simplification for this particular sentence 

> With default values in `createStore()`, CoRedux starts main reducer coroutine ("manager"), when first `StateReceiver` subscribes to `CoreduxStore` instance.

be something like :

> Internally the reducer runs on its own coroutine. This ensures that only one action is processed by the reducer at one time and avoids concurrency problems from the very beginning.